pytest-selenium/pages/content.py

```diff
@@ -127,11 +127,50 @@ def click_previous_link(self):
     class NavBar(Region):
         _root_locator = (By.CSS_SELECTOR, '[data-testid="navbar"]')
         _openstax_logo_link_locator = (By.CSS_SELECTOR, "div > a")
+        _user_nav_locator = (By.CSS_SELECTOR, '[data-testid="user-nav"]')
+        _login_locator = (By.CSS_SELECTOR, "[data-testid='nav-login']")
+        _user_nav_toggle_locator = (By.CSS_SELECTOR, '[data-testid="user-nav-toggle"]')
+        _account_profile_locator = (By.XPATH, "//a[contains(text(), 'Account Profile')]")
+        _logout_locator = (By.XPATH, "//a[contains(text(), 'Log out')]")
 
         @property
         def openstax_logo_link(self):
             return self.find_element(*self._openstax_logo_link_locator).get_attribute("href")
 
+        @property
+        def user_nav(self):
+            return self.find_element(*self._user_nav_locator)
+
+        @property
+        def login(self):
+            return self.find_element(*self._login_locator)
+
+        @property
+        def user_is_not_logged_in(self):
+            return bool(self.login)
```

Comment:
If the `self.login` property isn't found, this will raise an exception instead of returning a `bool`. Use a `try` block to catch the `NoSuchElementException` possibility or `return bool(self.find_elements(*self._login_locator))` as `bool([])` (if the login isn't found) would evaluate as `False`.

Reply:
Done. But I have it like this with timeout exception:
```            try:
                self.wait.until(expected.visibility_of_element_located(self._login_locator))
                return bool(self.find_element(*self._login_locator))
            except TimeoutException:
                 return bool([])
```
because firefox was failing while looking for elements before the page loaded the usernav elements. Let me know if there is a better way